contracts/testhelpers/TestToken.sol

```diff
@@ -66,6 +66,8 @@ contract TestToken {
 		balances[_to] = balances[_to].add(_value);
 		if (allowanceAmount < uint256(-1)) {
 			allowed[_from][msg.sender] = allowanceAmount.sub(_value);
+			/// @dev Allowance mapping update requires an Approval event log
+			emit Approval(_from, msg.sender, _value);
```

Comment:
Same here - why Approval when this amount is subtracted from approved?

Reply:
The allowance has changed, so Approval event must be triggered, right? Maybe I should fix the parameter, instead of _value I should use the new allowance amount: allowed[_from][msg.sender] OR allowanceAmount.sub(_value)

Reply:
yes but now from the event it looks like approval was set to the _value which is actually subtracted from the allowance
so either event name should be changed or its parameters e.g. add _allowanceIncreased bool
```suggestion
			emit Approval(_from, msg.sender, _allowanceIncreased, _value);
```